adapters/minutemedia/minutemedia.go

```diff
@@ -0,0 +1,121 @@
+
+	"github.com/prebid/prebid-server/v2/adapters"
+	"github.com/prebid/prebid-server/v2/config"
+	"github.com/prebid/prebid-server/v2/openrtb_ext"
+)
+
+// adapter is a MinuteMedia implementation of the adapters.Bidder interface.
+type adapter struct {
+	endpointURL string
+}
+
+func Builder(_ openrtb_ext.BidderName, config config.Adapter, _ config.Server) (adapters.Bidder, error) {
+	return &adapter{
+		endpointURL: config.Endpoint,
+	}, nil
+}
+
+// MakeRequests prepares the HTTP requests which should be made to fetch bids.
+func (a *adapter) MakeRequests(openRTBRequest *openrtb2.BidRequest, _ *adapters.ExtraRequestInfo) (requestsToBidder []*adapters.RequestData, errs []error) {
+	org, err := extractOrg(openRTBRequest)
+	if err != nil {
+		errs = append(errs, fmt.Errorf("extractOrg: %w", err))
+		return nil, errs
+	}
+
+	openRTBRequestJSON, err := json.Marshal(openRTBRequest)
+	if err != nil {
+		errs = append(errs, fmt.Errorf("marshal bidRequest: %w", err))
+		return nil, errs
+	}
+
+	headers := http.Header{}
+	headers.Add("Content-Type", "application/json;charset=utf-8")
+
+	return append(requestsToBidder, &adapters.RequestData{
+		Method:  http.MethodPost,
+		Uri:     a.endpointURL + "?publisher_id=" + org,
+		Body:    openRTBRequestJSON,
+		Headers: headers,
+	}), nil
+}
+
+// MakeBids unpacks the server's response into Bids.
+func (a *adapter) MakeBids(request *openrtb2.BidRequest, _ *adapters.RequestData, responseData *adapters.ResponseData) (*adapters.BidderResponse, []error) {
+	if adapters.IsResponseStatusCodeNoContent(responseData) {
+		return nil, nil
+	}
+
+	if err := adapters.CheckResponseStatusCodeForErrors(responseData); err != nil {
+		return nil, []error{err}
+	}
+
+	var response openrtb2.BidResponse
+	if err := json.Unmarshal(responseData.Body, &response); err != nil {
+		return nil, []error{err}
+	}
+
+	bidResponse := adapters.NewBidderResponseWithBidsCapacity(len(request.Imp))
+	bidResponse.Currency = response.Cur
+
+	var errs []error
+
+	for _, seatBid := range response.SeatBid {
+		for i, bid := range seatBid.Bid {
+			bidType, err := getMediaTypeForBid(bid)
+			if err != nil {
+				errs = append(errs, err)
+				continue
+			}
+
+			bidResponse.Bids = append(bidResponse.Bids, &adapters.TypedBid{
+				Bid:     &seatBid.Bid[i],
+				BidType: bidType,
+			})
+		}
+	}
+
+	return bidResponse, errs
+}
+
+func extractOrg(openRTBRequest *openrtb2.BidRequest) (string, error) {
+	var err error
+	for _, imp := range openRTBRequest.Imp {
+		var bidderExt adapters.ExtImpBidder
+		if err = json.Unmarshal(imp.Ext, &bidderExt); err != nil {
+			return "", fmt.Errorf("unmarshal bidderExt: %w", err)
+		}
+
+		var impExt openrtb_ext.ImpExtMinuteMedia
+		if err = json.Unmarshal(bidderExt.Bidder, &impExt); err != nil {
+			return "", fmt.Errorf("unmarshal ImpExtMinuteMedia: %w", err)
+		}
+
+		if impExt.Org != "" {
+			return strings.TrimSpace(impExt.Org), nil
+		}
+	}
+
+	return "", errors.New("no org supplied")
+}
```

Comment:
This error condition can be avoided if you specify a min length of 1 in the query parameters json schema.

Reply:
I assume you mean this check: 
```go
if impExt.Org != "" {
```
I added the `minLength` to the JSON schema and removed the ceck.

I still need a return at the end of the function to satisfy the compiler so I changed the error message to `no imps in bid request` even though I believe that case is not possible (it's not possible for `MakeRequests` to be called for a bid request with no imps, right?)